docs/configuration/http_conn_man/route_config/route.rst

```diff
@@ -338,6 +342,27 @@ header_name
   the request header is not present, the load balancer will use a random number as the hash,
   effectively making the load balancing policy random.
 
+.. _config_http_conn_man_route_table_route_add_req_headers:
+
+Adding custom request headers
+-----------------------------
+
+Custom request headers can be added to a request that matches a specific route. The headers are
+specified in the following form:
+
+.. code-block:: json
+
+  [
+    {"key": "header1", "value": "value1"},
+    {"key": "header2", "value": "value2"}
+  ]
+
+*Note:* In the presence of duplicate header keys, headers specified at the
+individual route-level take precedence over those specified at 
```

Comment:
This is not exactly clear: For headers that are not in the O(1) set, the header *will* get appended multiple times. So there is precedent, but only in the ordering. This is kind of complicated, and not sure whether you want to try to describe this in the docs, or just rephrase to talk about the order of addition.